src/sys/unix.rs

```diff
@@ -44,27 +54,32 @@ pub fn set_terminal_attr(termios: &Termios) -> Result<()> {
 }
 
 pub fn enable_raw_mode() -> Result<()> {
-    let mut ios = get_terminal_attr()?;
-    let prev_ios = ios;
-
-    unsafe {
-        if ORIGINAL_TERMINAL_MODE.is_none() {
-            ORIGINAL_TERMINAL_MODE = Some(prev_ios.clone());
-        }
+    let mut original_mode = TERMINAL_MODE_PRIOR_RAW_MODE.lock().unwrap();
```

Comment:
Can't we use `?` instead of unwrap?

Reply:
Doesn't make sense. The only error here is `PoisonError`, which is described as:

Poisoning:

> The mutexes in this module implement a strategy called "poisoning" where a mutex is considered poisoned whenever a thread panics while holding the mutex. Once a mutex is poisoned, all other threads are unable to access the data by default as it is likely tainted (some invariant is not being upheld).

`lock()`:

> If another user of this mutex panicked while holding the mutex, then this call will return an error once the mutex is acquired.

If this happens, I'd rather crash, because these data are a) already unaccessible, b) the whole program is in a weird state.

Reply:
Agree, then we should keep it this way